### PR TITLE
A11y - Add Pa11y CI in main pipeline to collect a11y issues

### LINKED ISSUE #36555: Accessibility checks -  Add Pa11y CI in main pipeline to collect a11y issues
In order to improve accessibility in Grafana, there is a need to continuously check how accessible our UI layout is. This would help us track how close we are to being WCAG compliant. Outlined below are some of the approaches we hope to take in order to achieve our goal.

**Todos**:

- [x] Research on how we can add pa11y CI to our yarn task
- [x] Experiment with different approaches to see which one works best.
- [x] Fetch a11y metrics and populate on a dashboard CI stats

**Pages to test**: PR #36841

- [x] Dashboard landing page 
- [x] DashboardSettings Page
- [x] Search Page
- [x] Bar Gauge Panel Edit Page 
- TimeSeries Panel Edit page - not going to implement this on this iteration 
- [x] Explore page
- [x] Manage Dashboards page
- [x] Alerting list page
- [x] Configurations pages (data sources, users, plugins, preferences)

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

Collect a11y metrics for the main pages and send them to the Grafana CI metrics. Metrics are going to be collected every time we merge new code into the main branch, in the future, we are going to add this step in the PR stage.

This PR is collecting a11y issues from 13 pages:

1. Login
2. Home
3. Search
4. Panel Baur Gauge View
5. Panel Baur Gauge Edit
6. Alerting
7. Explore
8. Configuration: DataSources
9. Configuration: Users
10. Configuration: Teams
11. Configuration: Plugins
12. Configuration: Preferences
13. Configuration: API keys

**Which issue(s) this PR fixes**:

Fixes #36555

**Special notes for your reviewer**:

The a11y report is NOT being store anyware, we are only sending the number of issues to the CI dashboard

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
Significant work on wrapping this up, but I'm a bit confused or maybe I haven't understood the scope for this PR.

I thought we wanted to fail the build if the a11y errors were above a certain hard coded threshold also I think that the build fail should occur in PR builds as well so we get quicker feedback. (reporting should be on main branch only as you've done in the PR)

Lastly I don't think we should introduce this in 8.1 branch but target it for 8.2 milestone.

**Comment by axelavargas** [comment]
> Significant work on wrapping this up, but I'm a bit confused or maybe I haven't understood the scope for this PR.
> 
> I thought we wanted to fail the build if the a11y errors were above a certain hard coded threshold also I think that the build fail should occur in PR builds as well so we get quicker feedback. (reporting should be on main branch only as you've done in the PR)
> 
> Lastly I don't think we should introduce this in 8.1 branch but target it for 8.2 milestone.

Hey Hugo, thank you for the feedback 🙌🏾 , I have moved the milestone to 8.2

Regarding the scope of the PR, indeed, this PR is limited, we only collected and publish the a11y metrics when we merge to the main branch.

For the PR stage we created another [issue,](https://github.com/grafana/grafana/issues/38063) there we are going to work on adding a11y checks on the PR pipeline. The reason behind this is because we still have some problems to solve.

For example, let's think on this scenario:

- Developer A: open a PR that is introducing 10 new a11y issues (above the threshold of 9), then the a11y step will fail and the CI will break preventing the merge
- With the current pa11y implementation, this person will go to the pipeline step and check what is the cause
- But when arrives it will see a huge log will all a11y issues in the terminal, making it really hard to identify what need to be fixed (a -1 for developer experience)

So we are thinking of some alternatives, for example, we could use something like GitHub actions, where we can comment on the PR ([POC](https://github.com/kaydelaney/pa11y-test/pull/2)), or we can maybe use a custom pa11y reporter, something like [this](https://gitlab-ci-utils.gitlab.io/pa11y-ci-reporter-html/) that is more readable.

**Comment by hugohaggmark** [comment]
> > Significant work on wrapping this up, but I'm a bit confused or maybe I haven't understood the scope for this PR.
> > I thought we wanted to fail the build if the a11y errors were above a certain hard coded threshold also I think that the build fail should occur in PR builds as well so we get quicker feedback. (reporting should be on main branch only as you've done in the PR)
> > Lastly I don't think we should introduce this in 8.1 branch but target it for 8.2 milestone.
> 
> Hey @hugohaggmark, thank you for the feedback 🙌🏾 , I have moved the milestone to 8.2
> 
> Regarding the scope of the PR, indeed, this PR is limited, we only collected and publish the a11y metrics when we merge to the main branch.
> 
> For the PR stage we created another [issue,](https://github.com/grafana/grafana/issues/38063) there we are going to work on adding a11y checks on the PR pipeline. The reason behind this is because we still have some problems to solve.
> 
> For example, let's think on this scenario:
> 
> * Developer A: open a PR that is introducing 10 new a11y issues (above the threshold of 9), then the a11y step will fail and the CI will break preventing the merge
> * With the current pa11y implementation, this person will go to the pipeline step and check what is the cause
> * But when arrives it will see a huge log will all a11y issues in the terminal, making it really hard to identify what need to be fixed (a -1 for developer experience)
> 
> So we are thinking of some alternatives, for example, we could use something like GitHub actions, where we can comment on the PR ([POC](https://github.com/kaydelaney/pa11y-test/pull/2)), or we can maybe use a custom pa11y reporter, something like [this](https://gitlab-ci-utils.gitlab.io/pa11y-ci-reporter-html/) that is more readable.

Awesome, I still think that the first iteration we should just have a threshold and the developer will have to fix any random a11y issues if the developer can't figure out what errors they introduced from the massive log of errors. 

Developers in OSS Grafana are accustomed to this behavior (doc errors and strict null errors work like this) so we shouldn't worry too much about that, the important thing is to prevent developers from introducing new a11y errors and to communicate that we are introducing this new threshold of course.

**Review by hugohaggmark — APPROVED** [review]
Perfect 🎉  and I agree we should add the threshold checks in a later PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+- **Dashboard:** Prevent incorrect panel layout by comparing window width against theme breakpoints. [#37868](https://github.com/grafana/grafana/pull/37868), [@ashharrison90](https://github.com/ashharrison90)
+- **Elasticsearch:** Fix metric names for alert queries. [#37871](https://github.com/grafana/grafana/pull/37871), [@dsotirakis](https://github.com/dsotirakis)
+- **Explore:** Fix showing of full log context. [#37442](https://github.com/grafana/grafana/pull/37442), [@ivanahuckova](https://github.com/ivanahuckova)
+- **PanelEdit:** Fix 'Actual' size by passing the correct panel size to Das…. [#37885](https://github.com/grafana/grafana/pull/37885), [@ashharrison90](https://github.com/ashharrison90)
+- **Plugins:** Fix TLS datasource settings. [#37797](https://github.com/grafana/grafana/pull/37797), [@wbrowne](https://github.com/wbrowne)
+- **Variables:** Fix issue with empty drop downs on navigation. [#37776](https://github.com/grafana/grafana/pull/37776), [@hugohaggmark](https://github.com/hugohaggmark)
+- **Variables:** Fix URL util converting `false` into `true`. [#37402](https://github.com/grafana/grafana/pull/37402), [@simPod](https://github.com/simPod)
```
Comment (hugohaggmark) [inline CHANGELOG.md:29]: This PR shouldn't change the CHANGELOG?

File: docs/sources/release-notes/release-notes-8-1-2.md [hunk 2]
```diff
@@ -10,31 +10,30 @@ list = false
...
+- **PanelEdit:** Fix 'Actual' size by passing the correct panel size to Das…. [#37885](https://github.com/grafana/grafana/pull/37885), [@ashharrison90](https://github.com/ashharrison90)
+- **Plugins:** Fix TLS data source settings. [#37797](https://github.com/grafana/grafana/pull/37797), [@wbrowne](https://github.com/wbrowne)
+- **Variables:** Fix issue with empty drop downs on navigation. [#37776](https://github.com/grafana/grafana/pull/37776), [@hugohaggmark](https://github.com/hugohaggmark)
+- **Variables:** Fix URL util converting `false` into `true`. [#37402](https://github.com/grafana/grafana/pull/37402), [@simPod](https://github.com/simPod)
 
 ### Plugin development fixes & changes
 
-* **Toolkit:** Fix matchMedia not found error. [#37643](https://github.com/grafana/grafana/pull/37643), [@zoltanbedi](https://github.com/zoltanbedi)
-
+- **Toolkit:** Fix matchMedia not found error. [#37643](https://github.com/grafana/grafana/pull/37643), [@zoltanbedi](https://github.com/zoltanbedi)
```
Comment (hugohaggmark) [inline docs/sources/release-notes/release-notes-8-1-2.md:39]: This PR shouldn't change the release notes, right?

Comment (axelavargas) [reply docs/sources/release-notes/release-notes-8-1-2.md:39]: yes :(, I think I have some issues while merging main branch

**Comment by axelavargas** [comment]
@hugohaggmark I reverted the changes related to those `.md` files, but investigating the problem, it seems we [introduced](https://github.com/grafana/grafana/pull/37616) the ability to prettify md files on lint-staged (this happens when we want to commit.) In my case, because I needed to resolve conflicts it automatically formatted those files that were updated on main after the last release.

The reason why they needed to be formatted, is because the code to generate our CHANGELOG and release notes are using the [old prettier format for unordered list](https://github.com/grafana/grafana-github-actions/blob/main/update-changelog/ReleaseNotesBuilder.ts#L215) `*`, now prettier uses `-`. I found a GitHub [discussion](https://github.com/prettier/prettier/issues/5308) that was talking about this.


I will open a PR on the grafana gitHub actions repo to fix this.

**Pull request merged**